### PR TITLE
modified publish workflow

### PR DESCRIPTION
**Pull request by kecnry**
This replaces #949, and now adapts the matrix wheel generation used by https://github.com/aprsa/ndpolator/blob/main/.github/workflows/on_release.yaml

### CODE REVIEW

**Comment by kecnry** [comment]
looks like there is a bug with building the wheel for libphoebe:

```
building 'libphoebe' extension
        creating build/temp.linux-x86_64-pypy38/phoebe/lib
        g++ -pthread -DNDEBUG -O2 -fPIC -I/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/numpy/core/include -I/opt/python/pp38-pypy38_pp73/include/pypy3.8 -c phoebe/lib/libphoebe.cpp -o build/temp.linux-x86_64-pypy38/phoebe/lib/libphoebe.o -std=c++14
        Traceback (most recent call last):
          File "/opt/python/pp38-pypy38_pp73/lib/pypy3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
            main()
          File "/opt/python/pp38-pypy38_pp73/lib/pypy3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
            json_out['return_val'] = hook(**hook_input['kwargs'])
          File "/opt/python/pp38-pypy38_pp73/lib/pypy3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
            return _build_backend().build_wheel(wheel_directory, config_settings,
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/build_meta.py", line 421, in build_wheel
            return self._build_with_temp_dir(
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/build_meta.py", line 403, in _build_with_temp_dir
            self.run_setup()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/build_meta.py", line 318, in run_setup
            exec(code, locals())
          File "<string>", line 20, in <module>
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/__init__.py", line 117, in setup
            return distutils.core.setup(**attrs)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/core.py", line 183, in setup
            return run_commands(dist)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/core.py", line 199, in run_commands
            dist.run_commands()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/dist.py", line 954, in run_commands
            self.run_command(cmd)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/dist.py", line 950, in run_command
            super().run_command(command)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
            cmd_obj.run()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/command/bdist_wheel.py", line 398, in run
            self.run_command("build")
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
            self.distribution.run_command(command)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/dist.py", line 950, in run_command
            super().run_command(command)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
            cmd_obj.run()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/command/build.py", line 135, in run
            self.run_command(cmd_name)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
            self.distribution.run_command(command)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/dist.py", line 950, in run_command
            super().run_command(command)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
            cmd_obj.run()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/command/build_ext.py", line 98, in run
            _build_ext.run(self)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
            self.build_extensions()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 476, in build_extensions
            self._build_extensions_serial()
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 502, in _build_extensions_serial
            self.build_extension(ext)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/command/build_ext.py", line 263, in build_extension
            _build_ext.build_extension(self, ext)
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 581, in build_extension
            self.compiler.link_shared_object(
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/ccompiler.py", line 758, in link_shared_object
            self.link(
          File "/tmp/pip-build-env-dk7xap70/overlay/lib/pypy3.8/site-packages/setuptools/_distutils/unixccompiler.py", line 269, in link
            self.linker_exe
        TypeError: 'NoneType' object is not subscriptable (key slice(None, None, None))
        [end of output]
```

**Comment by aprsa** [comment]
Well, the good news is that I cannot reproduce this locally, i.e. wheel builds without issues on 24.04. But that means that we either need to dig deeper, or change the wheel runner. I'm looking into it.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,52 +1,57 @@
...
+              with:
+                name: cibw-sdist
+                path: dist/*.tar.gz
+
+    build-wheels:
+        name: Build wheels on ${{ matrix.os }}
+        runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-22.04, macos-13, macos-14]
```
Comment (aprsa) [inline .github/workflows/publish.yml:28]: As of Sep 2024, ubuntu-24.04 runner is supported; can you change it and see if that fixes the issue?

Comment (kecnry) [reply .github/workflows/publish.yml]: nope, same error 🤷‍♂️

**Pull request merged**